.github/workflows/docker.yaml

```diff
@@ -167,6 +200,7 @@ jobs:
           push: true
           cache-from: type=registry,ref=${{ env.GH_IMAGE }}
           cache-to: type=inline
+          no-cache: true
```

Comment:
This is not yet needed here, because we don't have a `.dockerignore` file in this repo (yet).

Reply:
If we use caching here without the .dockerignore files aren't we always caching and not re-building source?

Reply:
No. When building images, docker always considers the build _context_. If the context has changed, caching will be disabled automatically. With the .dockerignore file, I explicitly disable most of the (source) context to just focus on the package.xml files that are relevant for system packages to be installed. Usually, all source changes are considered.

Reply:
I see your point, but looking at the docker jobs of GHA it seems source is not built again even though the repository has changes (see https://github.com/ros-planning/moveit2/actions/workflows/docker.yaml). Am I missing something?

Reply:
Hm. This is indeed not expected!